website/pages/restaurants/[restaurant]/menus/[menu].tsx

```diff
@@ -0,0 +1,169 @@
+"use client";
+
+import supabase from "@/utils/supabase";
+
+import Image from "next/image";
+import React, { useEffect, useState } from "react";
+
+import { Autoplay, EffectFade } from "swiper/modules";
+import { Swiper, SwiperSlide } from "swiper/react";
+
+import "swiper/css";
+import "swiper/css/effect-fade";
+import { useRouter } from "next/router";
+import NotFound from "@/components/PageNotFound";
+import VideoPlayer from "@/components/VideoPlayer";
+import MenuDish from "@/components/SkeletonPlaceholders/MenuDish";
+import LayoutWithoutFooter from "@/components/Layout/layoutWithoutFooter";
+
+const RestaurantMenu = () => {
+  const router = useRouter();
+  const { restaurant, menu } = router.query;
+  const suffix = restaurant
+    ?.toString()
+    .substring(restaurant?.lastIndexOf("-") + 1);
+  const menuSuffix = menu?.toString().substring(menu?.lastIndexOf("-") + 1);
+
+  const [isDishesLoading, setIsDishesLoading] = useState(true);
+  const [menuData, setMenuData] = useState<any[]>([]);
+
+  useEffect(() => {
+    const fetchDishes = async () => {
+      if (suffix) {
+        try {
+          const { data: menusData, error } = await supabase
+            .from("menus")
+            .select("id, name")
+            .eq("restaurant_id", atob(suffix!));
+          // .eq("id", atob(menuSuffix!));
+
+          if (error) return error;
+
+          const dishes = await Promise.all(
+            menusData.map(async (menu) => {
+              const { data: dishData, error: dishError } = await supabase
+                .from("dishes")
+                .select(
+                  "id, name, description, price, images, video, video_thumbnail"
+                )
+                .eq("menu_id", menu.id);
+
+              if (dishError) {
+                throw dishError;
+              }
+
+              return {
+                ...menu,
+                dishes: dishData,
+              };
+            })
+          );
+
+          setMenuData(dishes);
+        } catch (error) {
+          console.error("Error fetching dishes data:", error);
```

Comment:
How it will show error page?

Reply:
menu defined as empty and when we got some error here, menu will be empty and we have set that, if menus are not there then show 404 page